jni/src/faiss_wrapper.cpp

```diff
@@ -409,6 +410,114 @@ jobjectArray knn_jni::faiss_wrapper::QueryIndex_WithFilter(knn_jni::JNIUtilInter
     return results;
 }
 
+jobjectArray knn_jni::faiss_wrapper::QueryBinaryIndex_WithFilter(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jlong indexPointerJ,
```

Comment:
will the abstraction on the query flow be added in the upcoming PRs?

Reply:
Lets do that.

Reply:
https://github.com/opensearch-project/k-NN/issues/1773